docs/writing-rules/experiments/symbolic-propagation.md

```diff
@@ -27,3 +27,24 @@ If we tried to match the pattern `pandas.DataFrame(...).index.set_value(...)` ag
 Symbolic propagation is a generalization of [constant propagation](/writing-rules/data-flow/constant-propagation/) that addresses this limitation. It enables Semgrep to perform matching modulo variable assignments. Thus, Semgrep is then able to match both `test1` and `test2` with the same simple pattern. This feature needs to be enabled explicitly via rule `options:` by setting `symbolic_propagation: true`.
 
 <iframe src="https://semgrep.dev/embed/editor?snippet=JeBP" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+## Limitations
+
+Currently, symbolic propagation does not cross branching boundaries, such as `if` clauses or loops. This helps avoid difficulties with managing the parse graph for the code. Consider the following Python code, adapted from the example shown above:
```

Comment:
@IagoAbal I'd particularly love your feedback on the best way to phrase this (or if it's even important to explain the reason).

Reply:
I would drop this (and it's not 100% correct anyways):
> This helps avoid difficulties with managing the parse graph for the code.

Reply:
That sounds good, I'll fix it up by ✂️ that out entirely.